python/pyarrow/parquet.py

```diff
@@ -961,6 +960,23 @@ def __init__(self, path_or_paths, filesystem=None, schema=None,
             filters = _check_filters(filters)
             self._filter(filters)
 
+    def equals(self, other):
+        if self.fs.__class__ != other.fs.__class__:
+            return False
+        for prop in ('paths', 'memory_map', 'pieces', 'partitions',
+                     'common_metadata_path', 'metadata_path',
+                     'common_metadata', 'metadata', 'schema',
+                     'split_row_groups'):
+            if getattr(self, prop) != getattr(other, prop):
+                return False
+        return True
+
+    def __eq__(self, other):
+        try:
+            return self.equals(other)
+        except TypeError:
```

Comment:
This will never raise TypeError, right? Instead, should perhaps be:
```python
def __eq__(self, other):
    if not isinstance(other, ParquetDataset):
        return NotImplemented
```

Reply:
Indeed, copied from extensions where `equals` have type annotation.